interpreter/cling/lib/Interpreter/Value.cpp

```diff
@@ -255,11 +265,45 @@ namespace cling {
     const clang::ASTContext& ctx = getASTContext();
     unsigned payloadSize = ctx.getTypeSizeInChars(getType()).getQuantity();
     m_Storage.m_Ptr = AllocatedValue::CreatePayload(payloadSize, dtorFunc,
-                                                    GetNumberOfElements());
+                                                GetNumberOfElements(getType()));
+  }
+
+  void Value::AssertTypeMismatch(const char* Type) const {
+    assert(isBuiltinType() && "Must be a builtin!");
+    const clang::BuiltinType *BT = getType()->castAs<clang::BuiltinType>();
+    clang::PrintingPolicy Policy = getASTContext().getPrintingPolicy();
+    (void) BT; (void) Policy;
```

Comment:
```suggestion
#ifndef NDEBUG
    const clang::BuiltinType *BT = getType()->castAs<clang::BuiltinType>();
    clang::PrintingPolicy Policy = getASTContext().getPrintingPolicy();
#endif
```
See (https://github.com/root-project/root/pull/11252/files#r983935178, https://github.com/root-project/root/pull/11252/files#r982421753).

Reply:
But we removed CPU cost in case that function is called with NDEBUG on :)